cmd/otel-agent/config/agent_config.go

```diff
@@ -60,11 +60,26 @@ func NewConfigComponent(ctx context.Context, uris []string) (config.Component, e
 	apiKey := string(ddc.API.Key)
 	// Set the global agent config
 	pkgconfig := pkgconfigsetup.Datadog()
+	if len(ddCfg) != 0 {
+		// if the configuration file path was supplied on the command line,
+		// add that first so it's first in line
+		pkgconfig.AddConfigPath(ddCfg)
+		// If they set a config file directly, let's try to honor that
+		if strings.HasSuffix(ddCfg, ".yaml") || strings.HasSuffix(ddCfg, ".yml") {
+			pkgconfig.SetConfigFile(ddCfg)
+		}
```

Comment:
what is `ddCfg` if not a yaml path ? A path to a directory ?

Reply:
Yes. This code is a pattern borrowed from other `cmd` processes.

Reply:
Could we change it ? I just remove this behavior from our config mock since having both yaml and path to a yaml file in the same variable is confusing and error prone. Especially if we call `AddConfigPath` with it.